src/components/organisms/SettingsChange.tsx

```diff
@@ -0,0 +1,303 @@
+    let hotTempSetting: number =
+      document.getElementById("hotTempSetting").value;
+    let coldTempSetting: number =
+      document.getElementById("coldTempSetting").value;
+    let zipcode: string = document.getElementById("zipcode").value;
+
+    const updatedUserSettings: {} = {
+      // comments contain original data
+      first_name: firstName ? firstName : userData.first_name, // Akemi
+      last_name: lastName ? lastName : userData.last_name, // Kimura
+      email: email ? email : userData.email, //test@test.com
+      phone_number: phoneNumber ? phoneNumber : userData.phone_number, //090-0000-0000
+      old_password: oldPassword ? oldPassword : userData.old_password, //secretPassword
+      new_password: newPassword ? newPassword : "secretPassword",
+    };
+
+    const updatedDeviceSettings: {} = {
+      device_name: deviceName ? deviceName : deviceData[0].device_name, //Roppongi_Device
+      temperature_upper_limit: hotTempSetting
+        ? hotTempSetting
+        : deviceData[0].temperature_upper_limit, //30.2
+      temperature_lower_limit: coldTempSetting
+        ? coldTempSetting
+        : deviceData[0].temperature_lower_limit, //27?
+      zip_code: zipcode ? zipcode : deviceData[0].zip_code, //1001701
+    };
+
+    api.put(`/settings/user/`, updatedUserSettings).then((res) => {
+      console.log(res.data);
+      location.reload()
+    });
+
+    api.put(`/settings/device/`, updatedDeviceSettings).then((res) => {
+      console.log(res.data);
+    });
+    setSettingsView("read settings")
+  };
+
+  console.log("CHANGE", setSettingsView);
+  console.log("CHANGE", deviceData);
+
+  return (
+    <>
+      <div className={styles.top}>
+        <div className={styles.top__inner}></div>
+        <h2>User Settings</h2>
+        <Fieldset>
+          <Label htmlFor="firstName">First Name</Label>
+          <Input id="firstName" placeholder={userData.first_name} />
+        </Fieldset>
+        <Fieldset>
+          <Label htmlFor="lastName">Last Name</Label>
+          <Input id="lastName" placeholder={userData.last_name} />
+        </Fieldset>
+        <Fieldset>
+          <Label htmlFor="email">Email</Label>
+          <Input id="email" placeholder={userData.email} />
+        </Fieldset>
+        <Fieldset>
+          <Label htmlFor="oldPassword">Old Password</Label>
+          <Input id="oldPassword" type="password" />
+        </Fieldset>
+        <Fieldset>
+          <Label htmlFor="newPassword">New Password</Label>
+          <Input id="newPassword" type="password" />
+        </Fieldset>
+        <Fieldset>
+          <Label htmlFor="phoneNumber">Phone Number</Label>
+          <Input id="phoneNumber" placeholder={userData.phone_number} />
+        </Fieldset>
+        <h2>Device Settings</h2>
+        <Fieldset>
+          <Label htmlFor="deviceName">Device Name</Label>
+          <Input id="deviceName" placeholder={deviceData[0].device_name} />
```

Comment:
You can use `selectedDeviceIndex` instead of `0`

Reply:
Yep it works now thank you!